crates/fluvio-smartmodule-derive/src/generator/look_back.rs

```diff
@@ -36,15 +36,20 @@ pub fn generate_look_back_smartmodule(func: &SmartModuleFn) -> TokenStream {
                 }
 
                 let base_offset = smartmodule_input.base_offset();
+                let base_timestamp = smartmodule_input.base_timestamp();
```

Comment:
Gotcha! Thanks for pointing this out, here I provided an approach to make the dynamic code generation support `SmartModuleRecord` and the regular `Record`: https://github.com/infinyon/fluvio/pull/3389/commits/9ed2503daa43e99aa7b45f71e45ec0d919259938

For `Aggregate` I did not provided an approach because the `SmartModuleAggregateInput` is a different type.

Reply:
I think that we should add that also for aggergate

Reply:
> I think that we should add that also for aggergate

You mean adding support for `SmartModuleAggregateInput` to this generator right?

https://github.com/infinyon/fluvio/pull/3389/commits/b2a9851db90a55ab5afa63ddcc1ea3e097afa145#diff-f51d9c38a5813e14e10f56dbb11aa32ebd66f9bb742f3ee45b75d8051ca06e4aL39